src/WebJobs.Extensions.ApiHub/Config/ApiHubConfiguration.cs

```diff
@@ -23,6 +24,17 @@ public class ApiHubConfiguration : IExtensionConfigProvider, IFileTriggerStrateg
 
         ApiHubLogger _logger;
 
+        private ConnectionFactory ConnectionFactory { get; }
+
+        /// <summary>
+        /// Creates a new instance of this class.
+        /// </summary>
+        /// <param name="connectionFactory">The factory used to create connections</param>
+        public ApiHubConfiguration(ConnectionFactory connectionFactory = null)
```

Comment:
Let's keep anything like ConnectionFactory internal. We've got some open tracking items for improving the connection/setting access across all the extensions, and having a shared mechanism for doing this. See https://github.com/Azure/azure-webjobs-sdk-extensions/issues/52

So it would be good not to introduce a new public type that we'll have to remove. However, the fact that you went through the effort to create this underscores the need we have for a central solution :) See how the other extensions are accessing settings and follow suit for now.